### PR TITLE
REST > Rename ChargeBoxID as ChargingStationID

### PR DESCRIPTION
**Pull request by Ninjeneer**

### CODE REVIEW

File: src/app/services/central-server.service.ts [hunk 1]
```diff
@@ -2653,16 +2653,16 @@ export class CentralServerService {
       );
   }
 
-  public chargingStationsUnlockConnector(chargeBoxId: string, connectorId: number): Observable<ActionResponse> {
+  public chargingStationsUnlockConnector(chargingStationID: string, connectorId: number): Observable<ActionResponse> {
     this.checkInit();
     const body = {
-      chargeBoxID: chargeBoxId,
+      chargingStationID,
```
Comment (OliveGerste) [inline src/app/services/central-server.service.ts:2659]: Is this parameter still required or used somewhere? It is given in the URL now right?

**Review by OliveGerste — CHANGES_REQUESTED** [review]

File: src/app/services/central-server.service.ts [hunk 2]
```diff
@@ -3082,7 +3082,7 @@ export class CentralServerService {
     return this.httpClient.post<ActionResponse>(
       `${this.restServerSecuredURL}/${ServerRoute.REST_CHARGING_STATIONS_REQUEST_OCPP_PARAMETERS}`,
       {
-        chargeBoxID: id,
+        chargingStationID: id,
```
[the remark this answers is not in the document]
Comment (OliveGerste) [reply src/app/services/central-server.service.ts:3085]: I just saw your PR in the backend, which fixes this. Should I wait for the PR there before finishing the review?

**Comment by jerome-benoit** [comment]
REST API endpoint change must: 
+ bump the API version; 
+ keep the previous version endpoint working with the previous API (idempotency). 

That's why I've asked to have versionning per endpoint also possible and it unfortunately was not implemented that way. Since the API is not used per external customers, having a fine grained versioning on 1 digit per endpoints on a global API version on one digit make sense to keep backward compatibility easily.

**Comment by LucasBrazi06** [comment]
> REST API endpoint change must:
> 
> * bump the API version;
> * keep the previous version endpoint working with the previous API (idempotency).
> 
> That's why I've asked to have versionning per endpoint also possible and it unfortunately was not implemented that way. Since the API is now used per external customers, having a fine grained versioning on 1 digit per endpoints on a global API version also on one digit make sense to keep backward compatibility easily.

We have almost no one using the API for now, I can warn them and manage without increasing the version.
Version per endpoint is too complex as well as handling sub-version.
So we can afford to do this king of change for the time being.

**Comment by jerome-benoit** [comment]
> We have almost no one using the API for now, I can warn them and manage without increasing the version.
> Version per endpoint is too complex as well as handling sub-version.

If the API is designed to handle it, it's simple. 

> So we can afford to do this king of change for the time being.

That will not last as soon as the API will be more widely used, a proper version handling of the API must be implemented to deal with changes in a backward compatible manner.

**Comment by LucasBrazi06** [comment]
> > We have almost no one using the API for now, I can warn them and manage without increasing the version.
> > Version per endpoint is too complex as well as handling sub-version.
> 
> If the API is designed to handle it, it's simple.
> 
> > So we can afford to do this king of change for the time being.
> 
> That will not last as soon as the API will be more widely used, a proper version handling of the API must be implemented to deal with changes in a backward compatible manner.

Let's keep it simple for the time being as I like.
If the API was widely used, I wouldn't have allowed this change.

**Pull request merged**